src/ring_theory/finiteness.lean

```diff
@@ -382,7 +382,7 @@ begin
     (mv_polynomial.map_surjective f.to_ring_hom hf_surj).comp (alg_equiv.surjective _),
     ideal.fg_ker_comp _ _ _ _ (alg_equiv.surjective _)⟩,
   { convert submodule.fg_bot,
-    exact ring_hom.ker_coe_equiv _, },
+    exact ring_hom.ker_coe_equiv (mv_polynomial.sum_alg_equiv R ι ι').to_ring_equiv },
```

Comment:
The unifier had issues here because the goal contained `(coe : alg_hom → ring_hom) sum_alg_equiv.to_alg_hom` which had to be unified with `(coe : ring_equiv → ring_hom) sum_alg_equiv.to_ring_equiv`. Apparently the new generic coercion to `ring_hom` deriving from `ring_equiv.ring_hom_class` makes this unification slightly too hard without this hint.

In any case, we should be able to fix the requirement for unification entirely by generalizing `ker` to a `ring_hom_class`.